src/app/app.ts

```diff
@@ -24,14 +26,14 @@ export interface App {
 }
 
 class AppImpl implements App {
-    call: AppCall
-    context: AppContext
+    call: AppCallRequest
+    context: AppContextWithBot
     values: AppCallValues
 
-    constructor(call: AppCall) {
+    constructor(call: AppCallRequest) {
         this.call = call;
-        this.context = call.context;
-        this.values = call.values;
+        this.context = call.context as AppContextWithBot;
```

Comment:
Not every `Call` has an actin user. How do we deal with that case?

Reply:
I created multiple concreate types with required fields and the request in that call function determines which to use.